### DISCUSSION TITLE
Events registration

### DISCUSSION THREAD

**Discussion by paneq** [RailsEventStore/rails_event_store#1155] [Ideas]
@andrzejkrzywda proposed the idea of registering events in RES. Syntax and API not known. Let's brainstorm.

```ruby
class OrderPlaced
end

event_store.register(OrderPlaced) do |order_placed|
  # order_placed would be a proxy or something like it...
  order_placed.subscribe(to: [OrderSummaryHandler])
end
```

The benefit could be that RES would know about all possible events.

**Comment by jandudulski** [comment]
Why do you need a block? I guess this could be as simple as:

```ruby
event_store.register(OrderPlaced, to: [OrderSummaryHandler, FooHandler, BarHandler.new("args")])
```

Questions:

* what's the benefit of knowing all events by RES?
* what if some events won't be registered this way?
* what if the event has 0 subscribers? do we want to register them too?
